hydromt_wflow/wflow.py

```diff
@@ -1606,11 +1631,175 @@ def write_states(self):
         # raise NotImplementedError()
 
     def read_results(self):
-        """Read results at <root/?/> and parse to dict of xr.DataArray"""
+        """Read results at <root/?/> and parse to dict of xr.DataArray/xr.Dataset"""
         if not self._write:
             # start fresh in read-only mode
             self._results = dict()
-        # raise NotImplementedError()
+
+        # Read gridded netcdf (output section)
+        nc_fn = self.get_config("output.path", abs_path=True)
+        if nc_fn is not None and isfile(nc_fn):
+            self.logger.info(f"Read results from {nc_fn}")
+            ds = xr.open_dataset(nc_fn, chunks={"time": 30})
+            # TODO ? align coords names and values of results nc with staticmaps
+            self.set_results(ds, name="output")
+
+        # Read scalar netcdf (netcdf section)
+        ncs_fn = self.get_config("netcdf.path", abs_path=True)
+        if ncs_fn is not None and isfile(ncs_fn):
+            self.logger.info(f"Read results from {ncs_fn}")
+            ds = xr.open_dataset(ncs_fn, chunks={"time": 30})
+            self.set_results(ds, name="netcdf")
+
+        # Read csv timeseries (csv section)
+        csv_fn = self.get_config("csv.path", abs_path=True)
+        if csv_fn is not None and isfile(csv_fn):
+            # Count items by csv.column
+            count = 1
+            # Loop over csv.column
+            for col in self.get_config("csv.column"):
+                header = col["header"]
+                # Column based on map
+                if "map" in col.keys():
+                    # Read the corresponding map and derive the different locations
+                    # The centroid of the geometry is used as coordinates for the timeseries
+                    map_name = self.get_config(f"input.{col['map']}")
+                    da = self.staticmaps[map_name]
+                    gdf = da.raster.vectorize()
+                    gdf.geometry = gdf.geometry.centroid
+                    gdf.index = gdf.value.astype(da.dtype)
+                    gdf.index.name = "index"
+                    # Read the timeseries
+                    usecols = [0]
+                    usecols = np.append(
+                        usecols, np.arange(count, count + len(gdf.index))
+                    )
+                    count += len(gdf.index)
+                    da_ts = open_timeseries_from_table(
+                        csv_fn, name=f'{header}_{col["map"]}', usecols=usecols
+                    )
+                    da = GeoDataArray.from_gdf(gdf, da_ts, index_dim="index")
+                # Column based on xy coordinates or reducer for the full model domain domain
+                else:
+                    # Read the timeseries
+                    usecols = [0]
+                    usecols = np.append(usecols, np.arange(count, count + 1))
+                    count += 1
+                    try:
```

Comment:
Why do we need the try/ except clause here? Is this something we could identify before trying open_timeseries_from_table or somethin we could adapt within that method?

Reply:
I think the problem is when non numeric column heads are used in the csv file, then the method cannot find a numeric index. I used the try/except to avoid repeating many lines for the different wflow csv output options. 
I also see the non numeric index issue when trying to open observed csv timeseries so maybe something we could still adapt in that method?